service/api_task.go

```diff
@@ -411,25 +439,40 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, taskQueueService model.
 				"host":    currentHost.Id,
 				"message": "skipping task because of disabled project",
 			})
-			// Dequeue the task so we don't get it on another iteration of the loop.
-			if err = taskQueue.DequeueTask(nextTask.Id); err != nil {
-				return nil, errors.Wrapf(err,
-					"error pulling task with id %s from queue for distro %s",
-					nextTask.Id, nextTask.DistroId)
-			}
+
+			grip.Warning(message.WrapError(taskQueue.DequeueTask(nextTask.Id), message.Fields{
+				"message":              "projectRef.Enabled is false, but there was an issue dequeuing the task",
+				"distro":               nextTask.DistroId,
+				"task_id":              nextTask.Id,
+				"host":                 currentHost.Id,
+				"spec_group":           spec.Group,
+				"spec_build_variant":   spec.BuildVariant,
+				"spec_version":         spec.Version,
+				"spec_project_id":      spec.ProjectID,
+				"spec_group_max_hosts": spec.GroupMaxHosts,
+			}))
+
 			continue
 		}
 
+		// UpdateRunningTask updates the running task in the host document
 		ok, err := currentHost.UpdateRunningTask(nextTask)
 		if err != nil {
 			return nil, errors.WithStack(err)
 		}
 		// Dequeue the task so we don't get it on another iteration of the loop.
-		if err = taskQueue.DequeueTask(nextTask.Id); err != nil {
-			return nil, errors.Wrapf(err,
-				"error pulling task with id %s from queue for distro %s",
-				nextTask.Id, nextTask.DistroId)
-		}
+		grip.Warning(message.WrapError(taskQueue.DequeueTask(nextTask.Id), message.Fields{
```

Comment:
Same comment: The return should not be removed.

Reply:
As discussed: we'll leave this behaviour as is, for now.